File: src/org/openbravo/retail/posterminal/master/BPCategory.java
/*
 ************************************************************************************
 * Copyright (C) 2012 Openbravo S.L.U.
 * Licensed under the Openbravo Commercial License version 1.0
 * You may obtain a copy of the License at http://www.openbravo.com/legal/obcl.html
 * or in the legal folder of this module distribution.
 ************************************************************************************
 */
package org.openbravo.retail.posterminal.master;

import java.util.ArrayList;
import java.util.List;

import javax.enterprise.inject.Any;
import javax.enterprise.inject.Instance;
import javax.inject.Inject;

import org.codehaus.jettison.json.JSONException;
import org.codehaus.jettison.json.JSONObject;
import org.openbravo.client.kernel.ComponentProvider.Qualifier;
import org.openbravo.mobile.core.model.HQLPropertyList;
import org.openbravo.mobile.core.model.ModelExtension;
import org.openbravo.mobile.core.model.ModelExtensionUtils;
import org.openbravo.retail.posterminal.ProcessHQLQuery;

public class BPCategory extends ProcessHQLQuery {
  public static final String bpcategoryPropertyExtension = "OBPOS_BPCategoryExtension";

  @Inject
  @Any
  @Qualifier(bpcategoryPropertyExtension)
  private Instance<ModelExtension> extensions;

  @Override
  protected List<String> getQuery(JSONObject jsonsent) throws JSONException {

    List<String> hqlQueries = new ArrayList<String>();

    HQLPropertyList regularBPCategoryHQLProperties = ModelExtensionUtils
        .getPropertyExtensions(extensions);

    hqlQueries
        .add("select "
            + regularBPCategoryHQLProperties.getHqlSelect()
            + "from BusinessPartnerCategory bpcat "
            + "where "
            + "bpcat.$readableClientCriteria AND "
            + "bpcat.$naturalOrgCriteria AND "
            + "(bpcat.$incrementalUpdateCriteria) AND (bpcat.$incrementalUpdateCriteria)) order by bpcat.name");

    return hqlQueries;
  }

  @Override
  protected boolean bypassPreferenceCheck() {
    return true;
  }
}

File: web/org.openbravo.retail.posterminal/js/model/bpcategory.js
/*
 ************************************************************************************
 * Copyright (C) 2012-2019 Openbravo S.L.U.
 * Licensed under the Openbravo Commercial License version 1.0
 * You may obtain a copy of the License at http://www.openbravo.com/legal/obcl.html
 * or in the legal folder of this module distribution.
 ************************************************************************************
 */

(function() {
  class BPCategory extends OB.MasterdataModelDefinition {
    constructor() {
      super();

      this._indices = [
        {
          indexName: 'name',
          keyPath: 'name',
          objectParameters: { unique: false }
        }
      ];
      this._endPoint = 'org.openbravo.retail.posterminal.master.BPCategory';
    }
  }
  OB.MasterdataController.registerModel(BPCategory);
})();
